### DISCUSSION TITLE
VehicleClass always returns as "Car" even when configured as "EmergencyVehicle"

### DISCUSSION THREAD

**Discussion by SafrasIqbal** [eclipse/mosaic#170] [Q&A] [answered]
Hi There,
I am trying to get the VehicleClass from the CAM VehicleAwarenessData class. I have configured set of Cars and EmergencyVehicle in mapping file. However, it seems always I get the vehicle class as "Car"  by default. I just wondering am I missing anything here to get configured value correctly from CAM messages. Thank you in advance for the help. 

Mapping:
![mapping](https://user-images.githubusercontent.com/18607009/149246057-04b998a0-e5d1-44e3-a6e0-27763bbc0ee8.png)

Code:
```
        final Cam Cam = (Cam) msg;

        EncodedPayload encodedMessage = Cam.getPayLoad();
        CamContent decodedCam = encodedMessage.decodePayload();

        if(decodedCam.getAwarenessData() instanceof VehicleAwarenessData ) {
        VehicleAwarenessData decodedAwarenessData = (VehicleAwarenessData) decodedCam.getAwarenessData();        
        getLog().infoSimTime(this, "Processing CAM message class: {} Heading : {}, Direction: {} , From : {}",decodedAwarenessData.getVehicleClass(),decodedAwarenessData.getHeading(),decodedAwarenessData.getDirection(),msg.getRouting().getSource().getSourceName());
        }
```

**Comment by kschrab** [comment, marked as answer]
Hi, this is indeed a bug which I fixed in pull request #172. I guess this will be merged soon. Do you use the latest release or did you build MOSAIC by yourself?

**Reply by SafrasIqbal** [reply]
Hi @kschrab , Thank you for the quick response. I think I have build the MOSAIC myself. Its great that the bug addressed and fixed already. 

Please advise, how I can easily integrate the fix, replace the Java file and rebuild or any other option? Thank you.

**Reply by kschrab** [reply]
As soon as my change has been reviewed and accepted (hopefully today), you could just update your clone, call `mvn install -DskipTests` and find in `bundle/target` the new mosaic bundle.

**Reply by SafrasIqbal** [reply]
Hi @kschrab , Thank you for the advise. updates to the latest build and all works fine. thanks you
